include/envoy/network/connection.h

```diff
@@ -184,7 +184,7 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   /**
    * @return the const SSL connection data if this is an SSL connection, or nullptr if it is not.
    */
-  virtual const Ssl::Connection* ssl() const PURE;
+  virtual const Ssl::ConnectionInfo* ssl() const PURE;
```

Comment:
Is it possible to remove this interface from `Connection` and move into `StreamInfo`? If the change will be too heavy just leave a TODO.

Reply:
I'll add a TODO, it seems to require changing a bunch of functions that pass the connection around